plugins/catalog-backend-module-incremental-ingestion/src/database/IncrementalIngestionDatabaseManager.ts

```diff
@@ -546,12 +560,28 @@ export class IncrementalIngestionDatabaseManager {
    * @param entities - DeferredEntity[]
    */
   async createMarkEntities(markId: string, entities: DeferredEntity[]) {
+    const refs = entities.map(e => stringifyEntityRef(e.entity));
+
     await this.client.transaction(async tx => {
+      const existingRefsArray = (
+        await tx<{ ref: string }>('ingestion_mark_entities')
+          .select('ref')
+          .whereIn('ref', refs)
+      ).map(e => e.ref);
+
+      const existingRefsSet = new Set(existingRefsArray);
+
+      const newRefs = refs.filter(e => !existingRefsSet.has(e));
+
+      await tx('ingestion_mark_entities')
+        .update('ingestion_mark_id', markId)
+        .whereIn('ref', existingRefsArray);
```

Comment:
The `existingRefArray` is derived from the `DeferredEntity` array passed to the function, which will only contain the entities received during the active burst. So (hopefully) it's naturally self-limiting enough to prevent a situation where tens of thousands of entity refs could be passed in.